spec.bs

```diff
@@ -8372,49 +8739,81 @@ To <dfn>batch and fetch trusted scoring signals</dfn> given a [=trusted scoring
     1. Let |key| be (|request|'s [=trusted scoring signals request/seller script fetcher=],
       |request|'s [=trusted scoring signals request/base url=], |request|'s [=trusted scoring
       signals request/seller experiment group id=], |request|'s [=trusted scoring signals request/
-      top level origin=]).
+      top level origin=], |request|'s [=trusted scoring signals request/signal coordinator=]).
     1. If |batcher|'s [=trusted scoring signals batcher/request map=][|key|] does not [=map/exist=],
       then [=map/set=] |batcher|'s [=trusted scoring signals batcher/request map=][|key|] to an
       [=list/is empty|empty=] [=list=].
     1. [=list/Append=] |request| to |batcher|'s [=trusted scoring signals batcher/request
       map=][|key|].
-  1. Some number of times, heuristically, select a |key| and a non-[=list/is empty|empty=] [=set/
-    subset=] of |batcher|'s [=trusted scoring signals batcher/request map=][|key|], called
-    |entriesToBatch|, such that [=checking if trusted scoring signals batch honors URL length
-    limit=] on |batcher| and |entriesToBatch| returns true:
-
-      Note: implementations are free to wait to collect more requests to merge (by leaving things
-      in the |batcher|'s [=trusted scoring signals batcher/request map=]), or send them
-      individually, but need to take into account the configured URL length limit if they do combine
-      requests. All entries need to be handled eventually.
-
-      1. [=list/Remove=] |entriesToBatch| from |batcher|'s [=trusted scoring signals batcher/
-        request map=][|key|].
-      1. If |batcher|'s [=trusted scoring signals batcher/request map=][|key|] [=list/is empty=],
-        then [=map/remove=] |key| from |batcher|'s [=trusted scoring signals batcher/request map=].
-      1. Let |fullSignalsUrl| be the result of [=building batched trusted scoring signals url=] for
-        |entriesToBatch|.
+  1. Some number of times, heuristically, select a |key| from |batcher|'s [=trusted scoring signals
+    batcher/request map=]:
+    1. If |key|'s [=trusted scoring signals request/signal coordinator=] is null:

```

Comment:
same here. Can we add helper algorithms (like one for V1 and one for V2) for steps inside if/else blocks?

Reply:
Will also move this to a new block after this PR is merged.